lib/rules/no-meta-replaced-by.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * @fileoverview Disallows the usage of `meta.replacedBy` property
+ */
+
+'use strict';
+
+const utils = require('../utils');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow rules to use a `meta.replacedBy` property',
+      category: 'Rules',
+      recommended: true,
```

Comment:
Making a rule `recommended` is a breaking change so we'll need to do this later.

```suggestion
      recommended: false,
```